src/controller/TypedReadCallback.h

```diff
@@ -80,7 +80,7 @@ class TypedReadAttributeCallback final : public app::ReadClient::Callback
         VerifyOrDie(!aPath.IsListItemOperation());
 
         VerifyOrExit(aStatus.IsSuccess(), err = aStatus.ToChipError());
-        VerifyOrExit(aPath.mClusterId == mClusterId && aPath.mAttributeId == mAttributeId, err = CHIP_ERROR_SCHEMA_MISMATCH);
+        VerifyOrExit((aPath.mClusterId == mClusterId || kInvalidClusterId == mClusterId) && (aPath.mAttributeId == mAttributeId || kInvalidAttributeId == mAttributeId), err = CHIP_ERROR_SCHEMA_MISMATCH);
```

Comment:
what issue is being fixed here?

Reply:
The exact match of path with requested path was blocking support of wildcards though the underlying mechanism was built to work with wildcards (multiple read results). Hence, it was modified to verify exact match or wildcard.
For manufacturer specific extensions, I would imagine that manufacturers might want to use wildcards to discover endpoints as well.

Reply:
Please separate this outside of the platform change here.